External/Plugins/ASCompletion/Completion/ASGenerator.cs

```diff
@@ -1459,7 +1418,23 @@ private static void ConvertToConst(ClassModel inClass, ScintillaControl Sci, Mem
             }
 
             MemberModel m = NewMember(suggestion, member, FlagType.Variable | FlagType.Constant | FlagType.Static);
-            m.Type = ASContext.Context.Features.numberKey;
+
+            var features = ASContext.Context.Features;
+
+            switch (style)
+            {
+                case 4:
+                    m.Type = features.numberKey;
+                    break;
+                case 6:
+                case 7:
+                    m.Type = features.stringKey;
+                    break;
+                default:
+                    m.Type = features.dynamicKey ?? features.objectKey;

```

Comment:
In Haxe, if the type is unknown it's better to not generate any type annotation.